### PR TITLE
impl(rest): add LogWrapper overloads using RestContext

### PR DESCRIPTION
**Pull request by scotthart**
Add LogWrapper calls that can be used with REST LROs. By using a `typename Context` parameter we avoid having to include anything specifically REST.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595)

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @coryan)

___
*[`google/cloud/internal/log_wrapper.h` line 268 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595#-NM16jnwgwGr3ify5R3Z-r1-268:-NMACpkV3eYRb0R8QWo9:b-uvsphf) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f8fce26bc6f49faa2e7a9cada29d7a899a46b5f3/google/cloud/internal/log_wrapper.h#L268)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

That comment seems out of date, since we create an ID a few lines before?

Also, this file is part of the `grpc_utils` library.  Whatever the consequences of that might be.

</blockquote></details>

Removed comment.

The `rest_protobuf_internal` lib already dependes on `grpc_utils`.
___
*[`google/cloud/internal/log_wrapper_test.cc` line 208 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595#-NM16jnwgwGr3ify5R3_-r1-208:-NMACx8p3wBnvSJQzl1N:bqmjwcz) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/f8fce26bc6f49faa2e7a9cada29d7a899a46b5f3/google/cloud/internal/log_wrapper_test.cc#L208)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

nit: blank line here.
</blockquote></details>

Added.

File: google/cloud/internal/log_wrapper.h [hunk 1]
```diff
@@ -225,6 +225,62 @@ Result LogWrapper(Functor&& functor, google::cloud::CompletionQueue& cq,
...
+    typename std::enable_if<IsFutureStatus<Result>::value, int>::type = 0>
+Result LogWrapper(Functor&& functor, google::cloud::CompletionQueue& cq,
+                  Context& context, Request const& request, char const* where,
+                  TracingOptions const& options) {
+  // Because this is an asynchronous request we need a unique identifier so
+  // applications can match the request and response in the log.
+  auto prefix = std::string(where) + "(" + RequestIdForLogging() + ")";
+  GCP_LOG(DEBUG) << prefix << " << " << DebugString(request, options);
+  auto response = functor(cq, context, request);
+  // Ideally we would have an ID to match the request with the asynchronous
```
Comment (coryan) [inline google/cloud/internal/log_wrapper.h:241]: The same thing with the comment is repeated here.

**Review by coryan — APPROVED** [review]
Just fix the additional comment and then merge.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @coryan)

___
*[`google/cloud/internal/log_wrapper.h` line 241 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10595#-NM16jnwgwGr3ify5R3Z-r2-241:-NMAVF7s4OqgbPm-BsQk:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1810e2bcb84bfcc1e2d346bb400d4599a2643379/google/cloud/internal/log_wrapper.h#L241)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

The same thing with the comment is repeated here.
</blockquote></details>

Done.

**Pull request merged**